sdk/samples/iot/hub/src/pnp_helper.h

```diff
@@ -0,0 +1,137 @@
+ * @param[in] component_name An optional component name if the telemetry is being sent from a
+ *                           sub-component.
+ * @param[out] mqtt_topic A buffer with sufficient capacity to hold the MQTT topic. If
+ *                        successful, contains a null-terminated string with the topic that
+ *                        needs to be passed to the MQTT client.
+ * @param[in] mqtt_topic_size The size, in bytes of \p mqtt_topic.
+ * @param[out] out_mqtt_topic_length __[nullable]__ Contains the string length, in bytes, of
+ *                                                  \p mqtt_topic. Can be `NULL`.
+ * @return #az_result
+ */
+az_result pnp_helper_get_telemetry_topic(
+    az_iot_hub_client const* client,
+    az_iot_hub_client_properties* properties,
+    az_span component_name,
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length);
+
+/**
+ * @brief Parse a JSON payload for a PnP component command
+ *
+ * @param[in] component_command Input JSON payload containing the details for the component command.
+ * @param[out] component_name The parsed component name (if it exists).
+ * @param[out] command_name The parsed command name.
+ */
+az_result pnp_helper_parse_command_name(
+    az_span component_command,
+    az_span* component_name,
+    az_span* command_name);
+
+/**
+ * @brief Build a reported property
+ *
+ * @param[in] json_buffer The span into which the json payload will be placed.
+ * @param[in] component_name The name of the component for the reported_property.
+ * @param[in] property_name The name of the property to which to send an update.
+ * @param[in] append_callback The user callback to invoke to add the property value.
+ * @param[in] context The user context which is passed to the callback.
+ * @param[out] out_span The #az_span pointer to the output json payload.
+ */
+az_result pnp_helper_create_reported_property(
+    az_span json_buffer,
+    az_span component_name,
+    az_span property_name,
+    pnp_append_property_callback append_callback,
+    void* context,
+    az_span* out_span);
+
+/**
+ * @brief Build a reported property
+ *
+ * @param[in] json_buffer The span into which the json payload will be placed.
+ * @param[in] component_name The name of the component for the reported_property.
+ * @param[in] property_name The name of the property to which to send an update.
+ * @param[in] append_callback The user callback to invoke to add the property value.
+ * @param[in] context The user context which is passed to the callback.
+ * @param[in] ack_code The return value for the reported property.
+ * @param[in] ack_version The ack version for the reported property.
+ * @param[in] ack_description The optional description for the reported property.
+ * @param[out] out_span The #az_span pointer to the output json payload.
+ */
+az_result pnp_helper_create_reported_property_with_status(
```

Comment:
```suggestion
az_result pnp_create_reported_property_with_status(
```
Do we really need to call these helper? It's not like our SDK API names have helper in it.

Reply:
I forgot "helper" needs to be removed from this entirely. The V1 has it removed and needs to be done here